feature/visited/src/main/kotlin/uk/govuk/app/visited/VisitedViewModel.kt

```diff
@@ -8,28 +8,37 @@ import kotlinx.coroutines.flow.asStateFlow
 import kotlinx.coroutines.launch
 import uk.govuk.app.analytics.AnalyticsClient
 import uk.govuk.app.visited.data.VisitedRepo
+import uk.govuk.app.visited.data.store.VisitedLocalDataSource
 import uk.govuk.app.visited.data.transformVisitedItems
 import uk.govuk.app.visited.ui.model.VisitedUi
 import java.time.LocalDate
 import javax.inject.Inject
 
-
 internal data class VisitedUiState(
-    val visited: Map<String, List<VisitedUi>>
+    var visited: Map<String, List<VisitedUi>>?,
+    var hasSelectedItems: Boolean = false,
+    var hasAllSelectedItems: Boolean = false
```

Comment:
Is having default values necessary here? I think I can only see one place where we rely on the bottom one so could just set it instead?

Reply:
Also, looking at my above comment, could all the var's here be val's which I believe would be best practice for properties in a data class

Reply:
Fixed up